psydac/feec/tests/test_commuting_projections_dual.py

```diff
@@ -0,0 +1,149 @@
+from psydac.feec.derivatives import Gradient_3D
+from psydac.feec.derivatives import Curl_3D
+from psydac.feec.derivatives import Divergence_3D
+
+from sympde.expr                    import LinearForm, integral      
+from sympde.topology                import Derham, element_of, Cube  
+from psydac.api.discretization import discretize
+from psydac.api.settings import PSYDAC_BACKENDS
+
+import numpy as np
+from sympy import sin, cos
+import pytest
+
+
+@pytest.mark.parametrize('Nel', [8, 12])
+@pytest.mark.parametrize('p', [2, 3])
+@pytest.mark.parametrize('bc', [True, False])
+def test_3d_commuting_pro_dual_1(Nel, p, bc):
+    # Test transpose div
+
+    fun1    = lambda xi1, xi2, xi3 : sin(xi1)*sin(xi2)*sin(xi3)
+    D1fun1  = lambda xi1, xi2, xi3 : cos(xi1)*sin(xi2)*sin(xi3)
+    D2fun1  = lambda xi1, xi2, xi3 : sin(xi1)*cos(xi2)*sin(xi3)
+    D3fun1  = lambda xi1, xi2, xi3 : sin(xi1)*sin(xi2)*cos(xi3)
+
+    Nel = [Nel]*3
+    p   = [p]*3
+    bc  = [bc]*3
+
+    # Side lengths of logical cube [0, L]^3
+    L = [2*np.pi, 2*np.pi , 2*np.pi]
+
+    domain = Cube('domain', bounds1=(0, L[0]), bounds2=(0,L[1]), bounds3=(0,L[2]))
+    derham = Derham(domain)
+    domain_h = discretize(domain, ncells=Nel, periodic=bc)
+    derham_h = discretize(derham, domain_h, degree=p)
+
+    v2 = element_of(derham.V2, name='v2')
+    v3 = element_of(derham.V3, name='v3')
+    div = Divergence_3D(derham_h.V2, derham_h.V3)
+
+    f2 = LinearForm(v2, integral(domain, D1fun1(*domain.coordinates)*v2[0] + D2fun1(*domain.coordinates)*v2[1] + D3fun1(*domain.coordinates)*v2[2]))
+    f3 = LinearForm(v3, integral(domain, fun1(*domain.coordinates) * v3))
+
+    u2 = discretize(f2, domain_h, derham_h.V2, backend=PSYDAC_BACKENDS['pyccel-gcc']).assemble()
+    u3 = discretize(f3, domain_h, derham_h.V3, backend=PSYDAC_BACKENDS['pyccel-gcc']).assemble()
```

Comment:
Please pass the argument `nquads = (N, N, N)` with `N=10` (for example) and check if the error decreases.

If I remember correctly, the same argument should be passed to the discretization of the de Rham sequence. See also #332.